### PR TITLE
Issue 30 uri schema

### PR DESCRIPTION
**Pull request by evtimo**

### CODE REVIEW

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java [hunk 1]
```diff
@@ -72,7 +81,7 @@ public T create(
 			final int batchSize
 	) throws OmgShootMyFootException, InterruptedException {
 		return (T) new com.emc.mongoose.storage.driver.pravega.PravegaStorageDriver<I, O>(
-				Constants.DEFAULT_CONTROLLER_URI, stepId, dataInput, storageConfig, verifyFlag, batchSize
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java]: the URI schema value should be under the "storageConfig" structure with the relative path "net-uri-schema", no additional argument is needed

File: src/main/resources/config-schema-storage-net.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
 			"tcpNoDelay": "boolean",
 			"timeoutMilliSec": "int32",
 			"ssl": "boolean",
+			"storage-net-uri-schema":"string",
```
Comment (akurilov) [inline src/main/resources/config-schema-storage-net.json]: the option is already inside the "storage" and "net" blocks, should be instead:
```
            "uri" : {
                "schema": "string"
            }
```

File: src/main/resources/config/defaults-storage-net.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
 			// Currently only HTTPS implementation is available.
 			// Have no effect if configured storage type is filesystem.
 			"ssl": false,
+			"storage-net-uri-schema": "tcp",
```
Comment (akurilov) [inline src/main/resources/config/defaults-storage-net.json]: the option is already inside the "storage" and "net" blocks, should be instead:
```
            "uri": {
                "schema": "tcp"
            }
```

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java [hunk 4]
```diff
@@ -55,13 +55,13 @@ public PravegaStorageDriver(
 			throws OmgShootMyFootException {
 		super(testStepId, dataInput, storageConfig, verifyFlag, batchSize);
 		this.uriSchema = uriSchema;
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java]: should be:
```
this.uriSchema = storageConfig.stringVal("net-uri-schema");
```

**Review by akurilov — CHANGES_REQUESTED** [review]

File: src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java [hunk 5]
```diff
@@ -51,6 +51,15 @@ public final String id() {
 		return NAME;
 	}
 
+	private final String uriSchema()  {
```
Comment (akurilov) [inline src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriverExtension.java]: `SCHEMA_PROVIDER` is a configuration validation schema. This doesn't relate to the URI schema. I don't think that this method is needed

**Review by akurilov — CHANGES_REQUESTED** [review]

**Review by veronikaKochugova — APPROVED** [review]

**Review by akurilov — APPROVED** [review]

**Comment by veronikaKochugova** [comment]
Please resolve the conflict

**Review by veronikaKochugova — CHANGES_REQUESTED** [review]
Please resolve the conflict

**Comment by evtimo** [comment]
tests fail because of missing parts of code from other new branches.
compilation without errors.

**Comment by evtimo** [comment]
Difficult to resolve conflicts after adoption, issue solved in another branch

**Pull request closed without merging**